dev/unix/release.sh

```diff
@@ -0,0 +1,135 @@
+#!/usr/bin/env bash
+# Script to build the binaries and package them up for release.
+# This should be run from the top-level directory.
+
+notion_info() {
+  local ACTION="$1"
+  local DETAILS="$2"
+  command printf '\033[1;32m%12s\033[0m %s\n' "${ACTION}" "${DETAILS}" 1>&2
+}
+
+notion_error() {
+  command printf '\033[1;31mError\033[0m: %s\n' "$1" 1>&2
+}
+
+bold() {
+  command printf '\033[1m%s\033[0m' "$1"
+}
+
+# parse the 'version = "0.3.0"' line from the input text
+parse_version() {
+  local contents="$1"
+
+  while read -r line
+  do
+    if [[ "$line" =~ ^version\ =\ \"(.*)\" ]]
+    then
+      echo "${BASH_REMATCH[1]}"
+      return 0
+    fi
+  done <<< "$contents"
+
+  notion_error "Could not determine the current version"
+  return 1
+}
+
+# returns the os name to be used in the packaged release,
+# including the openssl info if necessary
+parse_os_info() {
+  local uname_str="$1"
+  local openssl_version="$2"
+
+  case "$uname_str" in
+    Linux)
+      major_minor="$(parse_openssl_version "$openssl_version")"
+      # if there was an error, return
+      exit_code="$?"
+      if [ "$exit_code" != 0 ]
+      then
+        return "$exit_code"
+      fi
+
+      echo "linux-openssl-$major_minor"
+      ;;
+    Darwin)
+      echo "macos"
+      ;;
+    *)
+      notion_error "Releases for '$uname_str' are not yet supported. You will need to add another OS case to this script, and to the install script to support this OS."
+      return 1
+  esac
+  return 0
+}
+
+# return true(0) if the element is contained in the input arguments
+# called like:
+#  if element_in "foo" "${array[@]}"; then ...
+element_in() {
+  local match="$1";
+  shift
+
+  local element;
+  # loop over the input arguments and return when a match is found
+  for element in "$@"
+  do
+    [ "$element" == "$match" ] && return 0
+  done
+  return 1
+}
+
+# parse the OpenSSL version from the input text
+parse_openssl_version() {
+  local version_str="$1"
+
+  # array containing the SSL libraries that are supported
+  # would be nice to use a bash 4.x associative array, but bash 3.x is the default on OSX
+  SUPPORTED_SSL_LIBS=( 'OpenSSL' )
+
+  if [[ "$version_str" =~ ^([^\ ]*)\ ([0-9]+\.[0-9]+) ]]
```

Comment:
I'm not sure if the current release process supports this, but we probably need to support parsing the full 3-part version of OpenSSL, since `1.0.1`, `1.0.2`, `1.1.0`, and `1.1.1` are all distinct and incompatible with one another. Though we also need to make sure to drop any letters from the end of the version, as that appears to be how OpenSSL distinguishes patch versions e.g. `1.0.2b` (which do seem to be compatible with one another).

Reply:
Ah, good point - I'll update this to use all three digits. I'm also working to revamp the install process, and will change that to use all three digits as well.